internal/credentials/xds/handshake_info_test.go

```diff
@@ -300,3 +306,120 @@ func TestMatchingSANExists_Success(t *testing.T) {
+			expected: false,
+		},
+		{
+			desc:     "different identity providers",
+			hi1:      NewHandshakeInfo(nil, mockProvider1, nil, false),
+			hi2:      NewHandshakeInfo(nil, mockProvider2, nil, false),
+			expected: false,
+		},
+		{
+			desc:     "same providers and SAN matchers",
+			hi1:      NewHandshakeInfo(mockProvider1, mockProvider1, nil, false),
+			hi2:      NewHandshakeInfo(mockProvider1, mockProvider1, nil, false),
+			expected: true,
+		},
+		{
+			desc: "different SAN matchers",
+			hi1: NewHandshakeInfo(mockProvider1, mockProvider1, []matcher.StringMatcher{
+				matcher.StringMatcherForTesting(newStringP("foo.com"), nil, nil, nil, nil, false),
+			}, false),
+			hi2: NewHandshakeInfo(mockProvider1, mockProvider1, []matcher.StringMatcher{
+				matcher.StringMatcherForTesting(newStringP("bar.com"), nil, nil, nil, nil, false),
+			}, false),
+			expected: false,
+		},
+		{
+			desc: "same SAN matchers",
+			hi1: NewHandshakeInfo(mockProvider1, mockProvider1, []matcher.StringMatcher{
+				matcher.StringMatcherForTesting(newStringP("foo.com"), nil, nil, nil, nil, false),
+			}, false),
+			hi2: NewHandshakeInfo(mockProvider1, mockProvider1, []matcher.StringMatcher{
+				matcher.StringMatcherForTesting(newStringP("foo.com"), nil, nil, nil, nil, false),
+			}, false),
+			expected: true,
+		},
+		{
+			desc:     "different requireClientCert flag",
+			hi1:      NewHandshakeInfo(mockProvider1, mockProvider1, nil, true),
+			hi2:      NewHandshakeInfo(mockProvider1, mockProvider1, nil, false),
+			expected: false,
+		},
+		{
+			desc: "same rootProvider but different mockCertProvider state",
+			hi1: &HandshakeInfo{
+				rootProvider:      mockProvider1,
+				identityProvider:  mockProvider1,
+				sanMatchers:       nil,
+				requireClientCert: false,
+			},
+			hi2: &HandshakeInfo{
+				rootProvider:      &mockCertProvider{id: 1},
+				identityProvider:  mockProvider1,
+				sanMatchers:       nil,
+				requireClientCert: false,
+			},
+			expected: false,
+		},
+		{
+			desc:     "identical mockCertProvider instances",
+			hi1:      NewHandshakeInfo(mockProvider1, mockProvider1, nil, false),
+			hi2:      NewHandshakeInfo(mockProvider1, mockProvider1, nil, false),
+			expected: true,
+		},
+		{
+			desc: "same SAN matchers with different content",
+			hi1: NewHandshakeInfo(mockProvider1, mockProvider1, []matcher.StringMatcher{
+				matcher.StringMatcherForTesting(newStringP("foo.com"), nil, nil, nil, nil, false),
+			}, false),
+			hi2: NewHandshakeInfo(mockProvider1, mockProvider1, []matcher.StringMatcher{
+				matcher.StringMatcherForTesting(newStringP("foo.com"), nil, nil, nil, nil, false),
+				matcher.StringMatcherForTesting(newStringP("bar.com"), nil, nil, nil, nil, false),
+			}, false),
+			expected: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			got := test.hi1.Equal(test.hi2)
+			if got != test.expected {
+				t.Errorf("hi1.Equal(hi2) = %v; want %v", got, test.expected)
```

Comment:
got before want

Reply:
Above comments are addressed and pushed the changes.